### PR TITLE
ENH: NDFrame.axis_ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
Refactor out NDFrame methods that operate only on axes into an `obj.axis_ops` namespace.  I'd want to do this in conjunction with a deprecation of the methods themselves xref #52110

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@phofl updated to include more underscores

File: pandas/core/axis_ops.py [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+        >>> df
+                                          species
+        class      diet
+        Mammals    Omnivore                Humans
+                   Carnivore                 Dogs
+        Reptiles   Carnivore               Snakes
+
+        Let's reorder the levels of the index:
+
+        >>> df.reorder_levels(["diet", "class"])
```
Comment (mroeschke) [inline pandas/core/axis_ops.py:79]: Shouldn't these example show the new usage of `df.axis_ops.foo`?

**Review by mroeschke — COMMENTED** [review]
Does `axis_ops` render in the API reference docs?

File: doc/source/whatsnew/v0.18.1.rst [hunk 2]
```diff
@@ -190,6 +190,7 @@ Partial string indexing now matches on ``DateTimeIndex`` when part of a ``MultiI
 On other levels
 
 .. ipython:: python
+   :okwarning:
```
Comment (mroeschke) [inline doc/source/whatsnew/v0.18.1.rst:193]: Could you just change this to use `axis_ops.swaplevel`?

**Review by mroeschke — COMMENTED** [review]
Could you also ensure the API reference page advertises e.g. `DataFrame.axis_ops.foo` instead of `DataFrame.foo`?

File: doc/source/whatsnew/v0.18.1.rst (hunk 2 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v0.18.1.rst:193]: i dont think we'd want to change this in the old doc

Comment (mroeschke) [reply doc/source/whatsnew/v0.18.1.rst:193]: Okay could we use `code block` instead?

**Comment by jbrockmendel** [comment]
> Could you also ensure the API reference page advertises e.g. DataFrame.axis_ops.foo instead of DataFrame.foo?

I'm not clear on how that works.  For e.g. swaplevel would I remove the entry for Series.swaplevel in source/reference/series.rst? I could add Series.axis_ops in the same place but will adding Series.axis_ops.swap_level render?

**Comment by mroeschke** [comment]
> I could add Series.axis_ops in the same place but will adding Series.axis_ops.swap_level render?

I think so. I think we have an `autosummary` template that renders our accessors (like Series.str.foo)`

**Comment by mroeschke** [comment]
Yeah actually that's a good point. if there's an `index.foo()` alternative might as well deprecate the `DataFrame.foo()` behavior

**Comment by jbrockmendel** [comment]
Closing for lack of interest, will do individual deprecations in the subset of cases where that is feasible

**Comment by jbrockmendel** [comment]
Reopening, will merge main and push shortly.

**Comment by jorisvandenbossche** [comment]
FWIW, while I understand the reason for adding something like this, I would not be very enthusiastic about an `axis_ops` accessor (from a user design point of view, it feels rather awkward)

**Pull request closed without merging**